packages/utils/src/fallbackSubgraph.ts

```diff
@@ -73,18 +76,22 @@ export class FallbackSubgraph<T extends SdkLike> {
     private readonly maxLag: number,
     private readonly stallTimeout = 10_000,
   ) {
-    this.sdks = sdks.map(({ client, uri }) => ({
+    const getSubgraphName = (uri: string) => {
+      const split = uri.split("/");
+      return split[split.length - 1];
+    };
+    this.subgraphs = sdks.map(({ client, uri }) => ({
       client,
       record: {
-        synced: true,
-        latestBlock: -1,
-        syncedBlock: -1,
-        // Setting maxLag + 1 as default to ensure we don't use the subgraph in this current state
-        // by virtue of this metric (sync() must be called first).
-        lag: this.maxLag + 1,
         // Typically used for logging, distinguishing between which subgraph is which, so we can monitor
         // which ones are most in sync.
         uri: uri.replace("https://", "").split(".com")[0],
```

Comment:
`uri: uri.replace("https://", "").split(".com")[0],`

I think it could be updated later with regex so it will look better. especially for different types of domains such as https://thegraph.io.

Reply:
Wow, yes this is pretty bad we should not be using `.com` for things.